src/graphql/queries/colony.graphql

```diff
@@ -85,3 +85,9 @@ query GetColonyMemberInvite($id: ID!) {
     invitesRemaining
   }
 }
+
+subscription OnUpdateColony {
+  onUpdateColony {
+    ...Colony
```

Comment:
I've been trying for the last few hours to figure out a way to make it work without getting the full fragment, but I can't seem to get it to work. I think it needs to do this in order to refetch the contributors properly. When I took out the fragment and replaced it with just this:

```graphql
subscription OnUpdateColonyReputation {
  onUpdateColony {
    id
    lastUpdatedContributorsWithReputation
    reputation
  }
}
```

then the contributors were not updating properly when refetched. Their `reputation` items were updating, but not the `colonyReputationPercentage`. In order to get this to update I had to refetch the colony from the network (note: not from the cache, so I couldn't just use the refetchColony from the context provider as that has a 'cache-first' policy). And if I'm refetching the colony from the network like that anyway then I think it's better to just keep the fragment in the subscription.

How concerning is this fragment? And could you think of anything which may help? I'm a bit at a loss today but can take another look tomorrow if it's really important that it doesn't use the fragment.

Reply:
@davecreaser I think there's a race condition between the subscription getting new data and contributor's `colonyReputationPercentage` actually being updated in the DB. 

I tried the lighter fragment you posted and got the same result of reputation percentage not being updated.

Then I added a 2s delay to the `searchColonyContributors` query refetch and it started returning the updated percentage value:

https://github.com/user-attachments/assets/923705c4-d324-4cb0-bd3b-3dde917e1052

The fact that `searchColonyContributors` is an OpenSearch powered query could explain this, posting updated data to OpenSearch is probably an asynchronous task. Fetching the bigger fragment took more time, so it coincidentally fixed the symptom. 

I added some logs to the lambda and could see the colony is updated (triggering the subscription) before the updated contributor is posted to OpenSearch:
<img width="569" alt="image" src="https://github.com/user-attachments/assets/ee351f4d-a4f6-49f9-adc3-d4b604cca3b1">

My main concern with subscribing to the entire fragment is the fact that this subscription will be getting updates pretty frequently due to the way `updateContributorsWithReputation` works. A lot of redundant data would keep getting sent, and because some of the colony fragment fields are resolved by lambdas, it could also end up increasing the AWS bill.

Setting a timeout to refetch the search contributors query feels a bit hacky, but I couldn't think of another solution to account for the delay, other than reverting to traditional polling for that query.